engine/renderer/public/gpu_resources.hpp

```diff
@@ -231,7 +231,8 @@ struct Mesh
         uint32_t indexOffset;
         float boundingRadius;
 
-        ResourceHandle<Material> material;
+        // Todo: change this into index into mesh material slots?
```

Comment:
I think this should stay a resource handle, i think the indices are more error prone